src/test/java/it/gov/pagopa/reward/notification/event/consumer/IbanOutcomeRecoveryIntegrationTest.java

```diff
@@ -0,0 +1,268 @@
+
+        waitForIbanStoreChanged(totalIban, rewardIbanRepository);
+
+        for (RewardsNotification r : rewardsNotificationList) {
+            RewardsNotification recovered = rewardsNotificationRepository.findById(r.getId()).block();
+            Assertions.assertNotNull(recovered);
+
+            int i = getIndexFromNotificationId(recovered.getId());
+            if (i >= 0 && i <= 3) { // status ERROR rejectionReason IBAN_NOT_FOUND - [0,3]
+                checkRecoveredErrorStatus(i, recovered);
+            }
+            else if (i >= 4 && i <= 7){ // status ERROR rejectionReason null - [4,7]
+                // the recovered notification should be identical to the r one, since it should be ignored by the RecoveryService
+                Assertions.assertEquals(r, recovered);
+            }
+
+            else if (i >= 8 && i <= 15) { // COMPLETED_KO - [8,15]
+                checkRecoveredCompletedKoStatus(i, r, recovered);
+            }
+
+            else {
+                throw new IllegalStateException("[IBAN_RECOVERY_TEST] Unexpected index: %d".formatted(i));
+            }
+
+        }
+    }
+
+    private IbanOutcomeDTO mockIbanOutcome(int bias) {
+        return IbanOutcomeDTO.builder()
+                .userId(USERID.formatted(bias))
+                .initiativeId(INITIATIVEID.formatted(bias))
+                .iban("IBAN%s".formatted(bias))
+                .status(IbanConstants.STATUS_OK)
+                .build();
+    }
+
+    private String getRemedialNotificationId(String id) {
+        String RECOVERY_ID_SUFFIX = "_recovery-";
+
+        String[] idSplit = id.split(RECOVERY_ID_SUFFIX);
+        if (idSplit.length == 2) {
+            int nextRecoveryProgressiveId = Integer.parseInt(idSplit[1]) + 1;
+            return "%s%s%d".formatted(idSplit[0], RECOVERY_ID_SUFFIX, nextRecoveryProgressiveId);
+        } else {
+            return "%s%s%d".formatted(id, RECOVERY_ID_SUFFIX, 1);
+        }
+
+    }
+
+    private int getIndexFromNotificationId(String id) {
+        String idWithoutUserId = id.substring(6);
+        String[] idSplit = idWithoutUserId.split("_[a-zA-Z]");
+        return Integer.parseInt(idSplit[0]);
+    }
+
+    private void checkRecoveredErrorStatus(int i, RewardsNotification recovered) {
+        Assertions.assertEquals(RewardNotificationStatus.TO_SEND, recovered.getStatus(), "notification %s".formatted(recovered.getId()));
+        Assertions.assertNull(recovered.getRejectionReason());
+        Assertions.assertNull(recovered.getResultCode());
+        Assertions.assertNull(recovered.getExportDate());
+
+        checkNotificationDate(i, recovered.getNotificationDate(), recovered.getId());
+    }
+
+    private void checkRecoveredCompletedKoStatus(int i, RewardsNotification r, RewardsNotification recovered) {
+        Assertions.assertEquals(getRemedialNotificationId(r.getId()), recovered.getRemedialId());
+        Assertions.assertEquals(RewardNotificationStatus.RECOVERED, recovered.getStatus());
```

Comment:
in order to underline the common purpose of these lines, move them into a separate checkRecovederNotification